modules/kuberay/main.tf

```diff
@@ -0,0 +1,61 @@
+# Configure a csi-driver based on a filestore, until mk8s will include a built-in csi-driver
+locals {
+  manifest_dir = "../modules/kuberay/csi-filestore/deploy"
+  yaml_files = fileset(local.manifest_dir, "*.yaml")
+  
+  # Read and split all YAML files
+  yaml_contents = [for f in local.yaml_files : split("---", file("${local.manifest_dir}/${f}"))]
+  
+  # Flatten the list of lists into a single list of YAML documents
+  flat_yaml_contents = flatten(local.yaml_contents)
+  
+  # Remove any empty documents
+  cleaned_yaml_contents = [for doc in local.flat_yaml_contents : doc if length(regexall("^\\s*$", doc)) == 0]
+}
+
+# Running 'kubectl apply -f ./csi-filestore/deploy' to set csi-driver & dependencies
+resource "kubectl_manifest" "csi_filestore_v2" {
```

Comment:
You could use official '[manifest](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest)' resource from hashicorp here, to prevent pulling new dependency.

Reply:
Since this solution wasn't that stable, I took mk8s helm release solution and integrated it in our terraform k8s-training.

It will be part of my next commit + other fixes